src/org/omegat/gui/matches/MatchesVarExpansion.java

```diff
@@ -244,6 +248,21 @@ public String expandVariables(NearString match) {
         localTemplate = localTemplate.replace(VAR_TARGET_TEXT, match.translation);
         localTemplate = localTemplate.replace(VAR_FUZZY_FLAG,
                 match.fuzzyMark ? (OStrings.getString("MATCHES_FUZZY_MARK") + " ") : "");
+
+        if (match.props != null) {
+            for (TMXProp prop : match.props) {
+                if (prop.getType().equals(ExternalTMFactory.TMXLoader.PROP_SOURCE_LANGUAGE)) {
+                    localTemplate = localTemplate.replace(VAR_SOURCE_LANGUAGE, prop.getValue());            
+                }
+                else if (prop.getType().equals(ExternalTMFactory.TMXLoader.PROP_TARGET_LANGUAGE)) {
```

Comment:
Can you put this `else` on the previous line? (`} else if (...`)

Reply:
> I think Kos asked for a way to disable this feature. Do you not want to do that, or do you consider the match penalty to be a substitute for an on/off switch?

The old behavior is still there so it doesn't bother Kos' use case (setting the penaly to 0 would be exactly like the current behavior, but it's not a very sensible default).
And I don't think there's a point letting the user disable multiple hits for their current target language.

Reply:
`The `else if` is fixed.

Reply:
So in the end, I included the option to set the score penalty and enable/disable the foreign match in the preferences. It made more sense because translators won't always be able to really make use the foreign language anyway.